### PR TITLE
Milestone 2

### PR DESCRIPTION
**Pull request by helman101**
## Friendships v1 (time est. 7 h)
- Create a Friendship model with the requestor, requester, and status columns
- Associate Friendship to User
- User can send and receive a Friend request
- User can reject or accept a Friend request
- User can see only friends posts in his Timeline

### CODE REVIEW

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## Hi @helman101 :wave: 

You've done well so far :+1:  and I commend you for that :clap: 
However, there are still some changes to be made before this milestone can be approved.

## REQUIRED

* [x] Timeline posts should include posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
  - As a logged in user, I can only see the post of the other user (who is not a friend) in my timeline. My posts are also not displayed. Kindly fix that.
* [x] There is a button to invite user to friendship on users list page and on single user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
You only have the button on the users list page.
* [ ] Add **Unit and Integration tests!** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

## NICE TO HAVE
* [x] The `Friend Request` button _Send Request_ or _Request Friendship_

### Kindly make the requested changes and submit for another review.
## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
### **STATUS: CHANGES REQUESTED🛑🛑**
Hello team, 
Great work implementing the changes. I love how the app is flowing.
However some improvements are needed.

[REQUESTED]: When you click on the profile of a user who is not your friend, you shouldn't see his post.
[REQUESTED]: When you post, you should see your own post on the timeline along posts of your friends.

Kindly make the required changes and ask for another review.👍

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED✔✔✔✔**
Hello team
Great work making the requested changes,
You can move to the following part.🙏

**Pull request merged**
